fre/make/gfdlfremake/buildBaremetal.py

```diff
@@ -120,10 +132,18 @@ def run(self):
         Param:
             - self : The dockerfile object
         """
-###### TODO make the Makefile
-        command = [self.bld+"/compile.sh","|","tee",self.bld+"/log.compile"]
-        try:
-            subprocess.run(args=command, check=True)
-        except:
-            print("There was an error running "+self.bld+"/compile.sh")
-            raise
+        command = [self.bld+"/compile.sh"]
+
+        # Run compile script
+        p1 = subprocess.Popen(command, stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+
+        # Direct output to log file as well
+        p2 = subprocess.Popen(["tee",self.bld+"/log.compile"], stdin=p1.stdout)
+
+        # Allow p1 to receive SIGPIPE is p2 exits
+        p1.stdout.close()
+        p2.communicate()
+
+        if p1.returncode != 0:
```

Comment:
This all looks good to me but for some reason when I tested this out I got the error message below on a successful build.

Reply:
Oh hmm, do you have the error message?

Reply:
Oh interesting, ok I'll check it out

Reply:
oh sorry I just meant like the message from the print statements on the two lines below:

https://github.com/NOAA-GFDL/fre-cli/blob/67ddb844d8fe505553431530cb3ad23ff95cb02c/fre/make/gfdlfremake/buildBaremetal.py#L148-L149

Not like a python runtime error or anything like that.

Reply:
I see, the return code check was not waiting for the first process to finish, so it was reading as `None`. I added `.wait()` before checking for the return code